File: Java codes/Lab test/src/BinarySearch.java
import java.util.*;
public class BinarySearch {

    public static int binarySearch(int[] array, int target) {
        int left = 0;
        int right = array.length - 1;

        while (left <= right) {
            int middle = left + (right - left) / 2;
            if (array[middle] == target) {
                return middle;
            }

            if (array[middle] < target) {
                left = middle + 1;
            }
            else {
                right = middle - 1;
            }
        }
        return -1;
    }
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int[] array = {1,2,3,4,5,6,7,8,9,10};
        System.out.print("Enter the number to search: ");
        int target = s.nextInt();
        int result = binarySearch(array, target);
        if (result == -1) {
            System.out.println("Number not found in the array.");
        } else {
            System.out.println("Number found at index: " + result);
        }

        s.close();
    }
}

File: 2-2/Algorithm/BinarySearch.cpp
#include<bits/stdc++.h>
using namespace std;

int binarySearch(vector<int>& v, int tar){
    int a = 0;
    int b = v.size() - 1;
    while(a < b){
        int m = a + (b - a) / 2;
        if(v[m] < tar){
            a = m + 1;
        }
        else{
            b = m;
        }
    }
    return (a < v.size() && v[a] == tar) ? a : -1;
}

int main(){
    vector<int>v = {1, 3, 5, 8, 10, 13, 17};
    sort(v.begin(),v.end());
    int target = 10;
    int result = binarySearch(v, target);
    if (result != -1) {
        cout << "Element found at index: " <<  result << endl;
    } else {
        cout << "Element not found in the array." << endl;
    }
}